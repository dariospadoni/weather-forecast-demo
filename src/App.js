import React, { useState, useEffect } from 'react';
import ForecastCard from './components/ForecastCard/ForecastCard';
import Summary from './components/Summary/Summary';
import LoadingSpinner from './components/LoadingSpinner/LoadingSpinner';
import './App.css';

/* using some public proxy to avoid CORS issues */
const WEATHER_DATA_URL = 'https://cors-anywhere.herokuapp.com/https://samples.openweathermap.org/data/2.5/forecast?q=M%C3%BCnchen,DE&appid=b6907d289e10d714a6e88b30761fae22';

function App() {
  const [loading, setLoading] = useState(true);
  const [city, setCity] = useState({});
  const [forecasts, setForecasts] = useState([]);
  const [summary, setSummary] = useState();
  const [selectedForecast, setSelectedForecast] = useState();

  const extractSummary = data => ({
    status: data.weather[0].main,
    date: data.dt_txt,
    tempMin: data.main.temp_min,
    tempMax: data.main.temp_max,
    temp: data.main.temp,
    icon: data.weather[0].icon,
  });

  const selectForecast = data => {
    setSelectedForecast(data.dt);
    setSummary(extractSummary(data));
    // smoothly scroll to top (useful for small screens)
    document.body.scrollIntoView({ behavior: 'smooth', block: 'start' });
  };

  useEffect(() => {
    fetch(WEATHER_DATA_URL)
      .then(response => response.json())
      .then(data => {
        setCity(data.city);
        setForecasts(data.list);
        setSelectedForecast(data.list[0].dt);
        setSummary(extractSummary(data.list[0]));
        setLoading(false);
      });
  }, []);

  return (
    <div className="App">

      {loading && <LoadingSpinner />}

      {!loading && <>
        
        <Summary weather={summary} cityName={city.name} />
        
        <ul className="forecastsList">
          {forecasts.map((d) =>
            <li key={d.dt} onClick={() => selectForecast(d)}  > 
              <ForecastCard
                selected={selectedForecast === d.dt}
                temperature={d.main.temp}
                date={new Date(d.dt_txt)} 
                icon={d.weather[0].icon} 
              />
            </li>)
          }
        </ul>
      </>}

    </div>
  );
}

export default App;
