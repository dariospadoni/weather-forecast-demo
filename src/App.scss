.App {
  min-height: 100vh;
  padding-top: 96px;
  
  & .forecastsList {
    display: flex;
    margin-top: 110px;
    overflow: scroll;
    list-style-type: none;
  }
}

@media (max-width: 640px) { 
  .App {
    padding-top: 48px;

    & .forecastsList {
      margin-top: 64px;
    }
  }
}
