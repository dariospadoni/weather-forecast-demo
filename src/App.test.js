import React from 'react';
import { render, waitForElementToBeRemoved, act, fireEvent } from '@testing-library/react';
import App from './App';
import mockData from '../__mocks__/weather.json';

let scrollIntoViewMock = jest.fn();
document.body.scrollIntoView = scrollIntoViewMock;

beforeEach(() => {
  jest.spyOn(global, 'fetch').mockResolvedValue({
    json: jest.fn().mockResolvedValue(mockData)
  })
});

afterEach(() => {
  jest.restoreAllMocks();
});

test('renders the loading indicator when no data has been fetched', async() => {
  await act(async () => {
    const { container } = render(<App />);
    expect(container.querySelector('progress')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
    expect(container.querySelector('header.summary')).not.toBeInTheDocument();
    expect(container.querySelector('ul.forecastsList')).not.toBeInTheDocument();
  });
});

test('renders correctly when data has been fetched', async() => {
  await act(async () => {
    const { container } = render(<App />);
    await waitForElementToBeRemoved(() => container.querySelector('progress'));
    expect(container).toMatchSnapshot();
    expect(container.querySelector('header.summary')).toBeInTheDocument();
    expect(container.querySelector('ul.forecastsList')).toBeInTheDocument();
  })
});

test('shows the selected forecast and scroll to top on click', async() => {
  await act(async () => {
    const { container } = render(<App />);
    await waitForElementToBeRemoved(() => container.querySelector('progress'));
    expect(container.querySelector('.summary__condition__temperature').textContent).toBe('14°');
    fireEvent.click(container.querySelector('ul.forecastsList > li:nth-child(4)'));
    expect(scrollIntoViewMock).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
    expect(container.querySelector('ul.forecastsList > li:nth-child(4) > article').classList.contains('forecast--active')).toBe(true);
    expect(container.querySelector('.summary__condition__temperature').textContent).toBe('0°');
  })
});
