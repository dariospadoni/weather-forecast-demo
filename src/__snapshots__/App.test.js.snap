// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when data has been fetched 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="summary"
    >
      <article
        class="summary__icon"
      >
        <svg>
          weather-sun.svg
        </svg>
      </article>
      <article
        class="summary__condition"
      >
        <div
          class="summary__overline summary__overline--status"
        >
          <div>
            Clear
          </div>
          <div>
            14°
             / 
            8°
          </div>
        </div>
        <div
          class="summary__condition__temperature"
        >
          14°
        </div>
      </article>
      <article>
        <div
          class="summary__overline"
        >
          Altstadt
        </div>
        <div
          class="summary__date"
        >
          Thursday
          <br />
          16. February
        </div>
      </article>
    </header>
    <section>
      <ul
        class="forecastsList"
      >
        <li>
          <article
            class="forecast__card forecast--active"
          >
            <div
              class="forecast__card__time"
            >
              12:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              14°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              15:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              13°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              18:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              4°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              21:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              0°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              00:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              0°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              03:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              06:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              3°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              09:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              4°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              12:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              15:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              3°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              18:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              21:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              00:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              03:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              06:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              1°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              09:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              12:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              15:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              18:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              1°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              21:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              00:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -4°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              03:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              06:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -6°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              09:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              0°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              12:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              4°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              15:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              18:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -1°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              21:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -4°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              00:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -4°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              03:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              06:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              09:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              1°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              12:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              6°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              15:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              6°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              18:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              1°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              21:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -1°
            </div>
          </article>
        </li>
      </ul>
    </section>
  </div>
</div>
`;

exports[`renders the loading indicator when no data has been fetched 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="spinner"
    >
      <label
        class="spinner__label"
      >
        Loading…
      </label>
      <progress />
    </div>
  </div>
</div>
`;

exports[`shows the selected forecast and scroll to top on click 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="summary"
    >
      <article
        class="summary__icon"
      >
        <svg>
          weather-sun.svg
        </svg>
      </article>
      <article
        class="summary__condition"
      >
        <div
          class="summary__overline summary__overline--status"
        >
          <div>
            Clear
          </div>
          <div>
            0°
             / 
            -2°
          </div>
        </div>
        <div
          class="summary__condition__temperature"
        >
          0°
        </div>
      </article>
      <article>
        <div
          class="summary__overline"
        >
          Altstadt
        </div>
        <div
          class="summary__date"
        >
          Thursday
          <br />
          16. February
        </div>
      </article>
    </header>
    <section>
      <ul
        class="forecastsList"
      >
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              12:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              14°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              15:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              13°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              18:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              4°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card forecast--active"
          >
            <div
              class="forecast__card__time"
            >
              21:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              0°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              00:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              0°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              03:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              06:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              3°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              09:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              4°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              12:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              15:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              3°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              18:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              21:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              00:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              03:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              06:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              1°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              09:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              12:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              15:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              18:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              1°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              21:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -2°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              00:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -4°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              03:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              06:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -6°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              09:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              0°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              12:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              4°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              15:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              18:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -1°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              21:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -4°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              00:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -4°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              03:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              06:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -5°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              09:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              1°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              12:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              6°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              15:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              6°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              18:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              1°
            </div>
          </article>
        </li>
        <li>
          <article
            class="forecast__card "
          >
            <div
              class="forecast__card__time"
            >
              21:00
            </div>
            <div
              class="forecast__card__condition"
            >
              <svg>
                weather-sun.svg
              </svg>
            </div>
            <div
              class="forecast__card__temperature"
            >
              -1°
            </div>
          </article>
        </li>
      </ul>
    </section>
  </div>
</div>
`;
